src/NServiceBus.Core/Unicast/Messages/LogicalMessage.cs

```diff
@@ -32,6 +32,7 @@ internal LogicalMessage(MessageMetadata metadata, object message, Dictionary<str
         /// <param name="newInstance">The new instance.</param>
         public void UpdateMessageInstance(object newInstance)
         {
+            Guard.AgainstDefault(newInstance, "newInstance");
```

Comment:
I think that depends on the user, they control it, not sure if our code
would work though

On 20 March 2015 at 13:41, Simon Cropp <notifications@github.com> wrote:

> In src/NServiceBus.Core/Unicast/Messages/LogicalMessage.cs
> <https://github.com/Particular/NServiceBus/pull/2686#discussion_r26817049>
> :
>
> > @@ -32,6 +32,7 @@ internal LogicalMessage(MessageMetadata metadata, object message, Dictionary<str
> >          /// <param name="newInstance">The new instance.</param>
> >          public void UpdateMessageInstance(object newInstance)
> >          {
> > +            Guard.AgainstDefault(newInstance, "newInstance");
>
> doesnt null only make sense for a control message?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/2686/files#r26817049>.
>